### PR TITLE
Add location selection page

### PR DESCRIPTION
**Pull request by harrygr**
Depends on https://github.com/lockeroom/lockeroom-server/pull/31

### CODE REVIEW

File: src/services/geocoding.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import makeMapboxClient from '@mapbox/mapbox-sdk';
+import makeGeocoderService from '@mapbox/mapbox-sdk/services/geocoding';
+import { config } from '../config';
+import { Geocoder } from '../contexts/GeocoderContext';
+
+const mbxClient = makeMapboxClient({ accessToken: config.mapboxToken });
+const mbxGeocoder = makeGeocoderService(mbxClient);
+
+/**
+ * An implementation of a geocoder that uses Mapbox
```
Comment (harrygr) [inline src/services/geocoding.ts]: Abstract away all the things!

File: src/vendor/mapbox-sdk-js.d.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
+declare module '@mapbox/mapbox-sdk' {
```
Comment (harrygr) [inline src/vendor/mapbox-sdk-js.d.ts]: I nicked these typings off a gist somewhere as no-one has added them to DefinitelyTyped yet. They seem correct though!

File: src/components/Map.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import * as L from 'leaflet';
```
Comment (Jarrydcrawford) [inline src/components/Map.tsx]:
```suggestion
import * as Leaflet from 'leaflet';
```

Comment (harrygr) [reply src/components/Map.tsx]: fair enough. I just use the same style as Leaflet does as the standalone package attaches itself to the window under `L`

**Pull request merged**